### PR TITLE
fix(D1): numerical filters and aggregations + emulated referential actions

### PR DESCRIPTION
**Pull request by jkomyno**
This PR:
- fixes https://github.com/prisma/team-orm/issues/1065
- fixes https://github.com/prisma/team-orm/issues/1050

### CODE REVIEW

File: query-engine/driver-adapters/src/conversion/sqlite.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ pub fn value_to_js_arg(value: &quaint::Value) -> serde_json::Result<JSArg> {
         quaint::ValueType::Json(Some(s)) => JSArg::Value(s.to_owned()),
         quaint::ValueType::Bytes(Some(bytes)) => JSArg::Buffer(bytes.to_vec()),
         quaint::ValueType::Int32(Some(value)) => JSArg::SafeInt(*value),
+        quaint::ValueType::Int64(Some(value)) => JSArg::SafeInt(*value as i32),
```
Comment (jkomyno) [inline query-engine/driver-adapters/src/conversion/sqlite.rs:13]: Controversial fix:

**Review by SevInf — CHANGES_REQUESTED** [review]
I don't think we should merge this.
Yes, it fixes some of the failures, but it also silently [screws up every int that is larger than 32 bits](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9664d971c7d258ebf5894ca618305ac7).
Error is better than silently reading/writing garbage.

**Comment by jkomyno** [comment]
> I don't think we should merge this. Yes, it fixes some of the failures, but it also silently [screws up every int that is larger than 32 bits](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9664d971c7d258ebf5894ca618305ac7). Error is better than silently reading/writing garbage.

yes, that's indeed not desireable. Would you have any advice to bypass the string serialisation of `i64`?

**Comment by SevInf** [comment]
Copying what I wrote in Slack in conversation with @jkomyno:

What I am thinking the best course of action here: do this change in D1 adapter code, not in the engines, so we at least won’t affect up Turso.
- for that we need to pass value types to the adapter, probably, another vector, `argTypes` [here[(https://github.com/prisma/prisma-engines/blob/289ad74feeb732dff1b2ecbbf974b9404b87205b/query-engine/driver-adapters/src/queryable.rs#L70) and adding new corresponding enum, derived from `QuaintValue` (type part only.
So, we are keeping i64 serialization as string on the engine side.

Then in the D1 adapter:
- Check argument types before sending them to D1.
- If we see Int64 - cast string to number but ensure it is done safely without data loss.
-  If cast can not be done  (number does not fit into 56bit, above `MAX_SAFE_INTEGER` or below `MIN_SAFE_INTEGER`) - throw an error telling user exactly that.

**Pull request closed without merging**